pointgrey_camera_driver/src/nodelet.cpp

```diff
@@ -160,6 +160,18 @@ class PointGreyCameraNodelet: public nodelet::Nodelet
   void connectCb()
   {
     NODELET_DEBUG("Connect callback!");
+
+    // Waiting for exposure node (a prerequisite) to finish being started for cam8
+    if (frame_id_ == "cam8")
+    {
+      ROS_INFO_STREAM("Waiting for cam8 exposure node to finish starting...");
+      const bool ret = ros::service::waitForService("/dynamic_exposure/"+frame_id_+"/ready", 20000); // 20 seconds
+      if (!ret) {
+        ROS_ERROR_STREAM("Timed out waiting for "<<frame_id_<<" exposure node to start. Exiting");
+        return;
+      }
```

Comment:
I don't know much about this specific use case but in general printing an error without returning any kind of error code makes me wonder if there's something else you should be doing - There's an error but no way to know progratically that there was actually an error.
This doesn't look like the function actually exits - The function itself returns, and something else exits.

Could you return an error code (enum maybe) from this function then check for error at the callsite? My thought would be to throw and exception but there are lots of other ways you could handle this

Reply:
Thanks for reviewing, Jared! 

I am now throwing an exception, rather than returning in the event of timeout.